backend/example_bib_app/services.py

```diff
@@ -122,6 +126,46 @@ async def StreamBooksByIDList(self, request, context):
             book = await self.get_object(book_id)
             yield book
 
+    @grpc_action(
+        request=[{"name": "filename", "type": "string"},
+                 {"name": "book_id", "type": "string"},
+                 {"name": "data", "type": "bytes"}],
+        request_name="FileChunk",
+        response=[{"name": "success", "type": "bool"}],
+        response_name="UploadStatus",
+        request_stream=True,
+    )
+    async def UpdateImage(self, request, context):
+        print("File upload started")
+
+        try:
+            with io.BytesIO() as f:
+                async for file_chunk in request:
+                    book_id = file_chunk.book_id
+                    filename = file_chunk.filename
+                    print("Reading file chunk:", len(file_chunk.data))
+                    f.write(file_chunk.data)
+            
+                queryset = self.get_queryset()
+
+                book = await sync_to_async(queryset.get)(book_id=book_id) 
```

Comment:
book = await queryset.aget(book_id=book_id)

Reply:
actually if you use the lookupfield in the request (remplacing book_id by just id) you can directly do something like:

book = await self.aget_object()